packages/xdl/src/Versions.js

```diff
@@ -109,18 +109,36 @@ export async function canTurtleBuildSdkVersion(sdkVersion, platform) {
     return true;
   }
 
-  const turtleSdkVersions = await Api.turtleSdkVersionsAsync();
-  if (!turtleSdkVersions || !turtleSdkVersions[platform]) {
-    return true;
+  if (semver.valid(sdkVersion) == null) {
+    throw new XDLError(
+      ErrorCode.INVALID_VERSION,
+      `"${sdkVersion}" is not a valid version. Must be in the form of x.y.z`
+    );
   }
 
-  try {
-    const turtleSdkVersion = turtleSdkVersions[platform];
-    return semver.gte(turtleSdkVersion, sdkVersion);
-  } catch (e) {
+  const turtleSdkVersions = await Api.turtleSdkVersionsAsync();
+  const expoSdkVersions = await Api.sdkVersionsAsync();
+  const expoSdkVersion = _.values(
+    _.pickBy(
+      expoSdkVersions,
+      (props, version) => semver.major(version) === semver.major(sdkVersion)
```

Comment:
Have we ever had more than one field for a given major SDK version in the `expoSdkVersions` object?

Reply:
No.

Reply:
So why do you need this complicated logic? Isn't `expoSdkVersions.find` function enough?

Reply:
I am naive enough to thought we could use it someday.